articles/deploying-nodejs-web-app/index.md

```diff
@@ -0,0 +1,143 @@
+sudo apt-get install python-certbot-nginx
+```
+
+Next, you need to add your domain to Nginx so Certbot can generate a certificate to the correct domain. Open the configuration file using `sudo nano /etc/nginx/sites-available/default` and replace the underscores in the server_name line to your domain. For example  `server_name yourdomain.com www.yourdomain.com;`. Save the file and exit by typing CTRL+X, y and then enter.
+
+To test, there are no errors in the file, type  `sudo nginx -t`  and if there’s none type  `sudo systemctl reload nginx`  to reload Nginx so it will use the updated configuration.
+
+Now we just need to generate the SSL certificate.  `sudo certbot --nginx -d yourdomain.com -d www.yourdomain.com` will start the process.  You should choose option 2 for the redirect process because it will forward anyone trying to access the insecure version of your site (http) to the secure (https) version instead.
+
+To test this, go to https://yourdomain.com and you should see the Nginx Welcome screen again.
+
+Finally, we're onto the last step, adding the Nginx configuration for your web app. Since it's easiest, we'll just modify the default one instead of creating a new one specifically for your web app. If you need to host several web apps on one droplet, you'd need to add a new configuration for each site.
+
+Type: `sudo nano /etc/nginx/sites-available/nano/default` to edit the default configuration file.
+
+You need to change the server_name parameter to the name of your domain. For example: yourdomain.com. Under location /, proxy_pass should be changed to http://localhost:(port name). The ssl_certificate_key should be modified: `/etc/letsencrypt/live/(domain name)/privkey.pem`. Finally, add the code block below to the end of the file and then type CTRL+X and then y to exit.
+
+```
+bash
+server {
+    if ($host = auroraspotter.space) {
+        return 301 https://$host$request_uri;
+    } # managed by Certbot
+        listen 80 default_server;
+        listen [::]:80 default_server;
+        server_name auroraspotter.space;
+    return 404; # managed by Certbot
+```
+To test, there are no errors in the file, type  `sudo nginx -t`  and if there’s none type  `sudo systemctl reload nginx`  to reload Nginx so it will use the updated configuration.
+Finally, you should be able to go to yourdomain.com and your web app will be running.
+
+### Running the App on Boot (Setting up a Process Manager)
+You've hooked your domain name up to your droplet and configured Nginx to serve your web app but how do you keep it running all the time especially after restarting your droplet? That's where a process manager comes in. It will manage your NodeJS web app, log any errors and start/stop it as needed. We will be using the process manager called PM2.
+The first step is to install PM2 using `sudo npm install pm2@latest -g`. Next, to run it on boot, use `pm2 startup systemd`. That will return the code you'll need to copy and paste into the terminal.
```

Comment:
You should show the reader what this looks like via a screenshot or a quote block to give them some idea of what the "code you'll need to copy and paste into the terminal" looks like. Also, it is not clear where or when they should copy and paste it into the terminal.